Algorithm.CSharp/BasicTemplateOptionsHistoryAlgorithm.cs

```diff
@@ -0,0 +1,112 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+using System;
+using System.Linq;
+using QuantConnect.Data;
+using QuantConnect.Data.Market;
+using QuantConnect.Orders;
+using QuantConnect.Data.UniverseSelection;
+using QuantConnect.Securities.Option;
+using QuantConnect.Securities;
+using QuantConnect.Indicators;
+
+namespace QuantConnect.Algorithm.CSharp
+{
+    /// <summary>
+    /// This example demonstrates how to get access to options history for a given underlying equity security.
+    /// </summary>
+    public class BasicTemplateOptionsHistoryAlgorithm : QCAlgorithm
+    {
+        private const string UnderlyingTicker = "GOOG";
+        public readonly Symbol Underlying = QuantConnect.Symbol.Create(UnderlyingTicker, SecurityType.Equity, Market.USA);
+        public readonly Symbol OptionSymbol = QuantConnect.Symbol.Create(UnderlyingTicker, SecurityType.Option, Market.USA);
+
+        public override void Initialize()
+        {
+            // this test opens position in the first day of trading, lives through stock split (7 for 1), and closes adjusted position on the second day
+            SetStartDate(2015, 12, 24);
+            SetEndDate(2015, 12, 24);
+            SetCash(1000000);
+
+            var equity = AddEquity(UnderlyingTicker);
+            var option = AddOption(UnderlyingTicker);
+
+            equity.SetDataNormalizationMode(DataNormalizationMode.Raw);
+            option.PriceModel = OptionPriceModels.CrankNicolsonFD();
+            //option.EnableGreekApproximation = true;
```

Comment:
This enables numerical greek approximation in case option pricing model itself doesn't produce them. This happens. I thought that if it is still possible to get an approximation for missing value, we should do it.